waveguide_mode_converter/mode_converter_meep_opt.py

```diff
@@ -334,80 +416,88 @@ def J(refl_mon,tran_mon):
 
     # initial guess for design parameters
     x = np.ones((n,)) * 0.5
+    x[Si_mask.flatten()] = 1    # set the edges of waveguides to silicon
+    x[SiO2_mask.flatten()] = 0  # set the other edges to SiO2
 
     # lower and upper bounds for design weights
     lb = np.zeros((n,))
+    lb[Si_mask.flatten()] = 1
     ub = np.ones((n,))
+    ub[SiO2_mask.flatten()] = 0
 
     # insert epigraph variable and bounds as first element of 1d arrays
-    x = np.insert(x, 0, 1.2)  # initial guess for the worst error (max: 2.0)
-    lb = np.insert(lb, 0, 0)  # lower bound: cannot be less than 0
-    ub = np.insert(ub, 0, 2)  # upper bound: cannot be more than 2
+    x = np.insert(x, 0, 1.2)   # initial guess for the worst error
+    lb = np.insert(lb, 0, 0.)  # lower bound: cannot be less than 0.
```

Comment:
Don't put any bounds on the dummy parameter. As @stevengj has said before, just let the optimizer do it's thing (especially during the inner iterations).